docs/sources/configure-client/grafana-agent/java/_index.md

```diff
@@ -0,0 +1,265 @@
+---
+title: "Profiling Java via the Grafana Agent"
+menuTitle: "Profiling Java via the Grafana Agent"
+description: "Learn about using Grafana Agent for continuous profiling Java processes for performance optimization."
+weight: 20
+---
+
+# Profiling Java via the Grafana Agent
+
+Grafana Agent supports Java profiling in [Flow mode](/docs/agent/latest/flow/). The configuration file is written in
+the [River](/docs/agent/latest/flow/config-language/) language and is composed of components that are used to collect,
+transform, and send data.
+
+## Configuration
+
+The `pyroscope.java` component is used to continuously profile Java processes running on the local Linux OS
+using [async-profiler](https://github.com/async-profiler/async-profiler).
+
+```river
+pyroscope.java "java" {
+  profiling_config {
+    interval = "15s"
+    alloc = "512k"
+    cpu = true
+    lock = "10ms"
+    sample_rate = 100
+  }
+  forward_to = [pyroscope.write.endpoint.receiver]
+  targets = discovery.relabel.java.output
+}
+```
+
+Using the `targets` argument, you can specify which processes and containers to profile on the machine. The `targets`
+can be from `discovery.process` component. You can use `discovery.process` join argument to join process targets with
+extra discoveries such as `dicovery.kubernetes`, `discovery.docker` and `discovery.dockerswarm`.
+To relabel discovered targets and set your own labels you can use the `discovery.relabel` component. For more
+information, see [Components](/docs/agent/latest/flow/concepts/components/).
+
+The `forward_to` parameter should point to a `pyroscope.write` component to send the collected profiles to your
+Pyroscope Server or [Grafana Cloud](/products/cloud/).
+
+| Name         | Type                     | Description                                      | Default | Required |
+|--------------|--------------------------|--------------------------------------------------|---------|----------|
+| `targets`    | `list(map(string))`      | List of java process targets to profile.         |         | yes      |
+| `forward_to` | `list(ProfilesReceiver)` | List of receivers to send collected profiles to. |         | yes      |
+| `tmp_dir`    | `string`                 | Temporary directory to store async-profiler.     | `/tmp`  | no       |
+
+The special label `__process_pid__` _must always_ be present in each target of `targets` and corresponds to the `PID` of
+the process to profile.
+
+The special label `service_name` is required and must always be present. If it is not specified, `pyroscope.java` will
+attempt to infer it from discovery meta labels. If `service_name` is not specified and could not be inferred, then it is
+set to `unspecified`.
```

Comment:
Clarifying what "it" refers to in some cases. The antecedent wasn't always clear. How does this read? 
```suggestion
The special label `service_name` is required and must always be present. If `service_name` is not specified, `pyroscope.java`
attempts to infer it from discovery meta labels. If `service_name` is not specified and could not be inferred, then it is
set to `unspecified`.
```